### ISSUE TITLE
[9.0.100-alpha.1.24070.3] dotnet tool install command get access deny message

### ISSUE THREAD

**Issue by Junjun-zhao** [dotnet/sdk#38183]
**Application Name**: globaltools
**OS**： Windows 10 21H2
**CPU**: X64
**.NET Build Number**: dotnet-sdk-9.0.100-alpha.1.24070.3
**Also can checking it on** https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1943809

**Verify Scenarios**:
1. Windows 10 21H2 x64 + dotnet-sdk-8.0.101: Pass
2. Windows 10 21H2 x64 + dotnet-sdk-9.0.100-alpha.1.24066.1: Pass 
3. Windows 10 21H2 x64 + dotnet-sdk-9.0.100-alpha.1.24070.3: Fail
4. Windows 10 21H2 x64 + dotnet-sdk-9.0.100-alpha.1.24072.15: Fail
5. Windows 10 21h2 x64 + dotnet-sdk-8.0.200-win-x64: Fail
6. Windows 10 21h2 x64 + dotnet-sdk-8.0.102-win-x64: Pass

**Issue Description**:
When installing dotnet tools with "gti -c install" command, it get access deny error for gti tool installation.

**Repro steps**:

**The machine only has dotnet-sdk-9.0.100-alpha.1.24070.3 installed. And environment variable DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX set as 2**.

1.Open command prompt window.
2. Execute "dotnet tool install –g gti" command.
4. Navigate to a local folder with "cd {localfolderpath}" command
7. Execute "gti –c save" command.
8. Execute "gti –c install" command.

**Expected Result**:
Tools will be installed successfully and "Tool is already installed" message will show if the tools were installed before.

![image](https://github.com/dotnet/sdk/assets/83067197/afec5e32-30eb-4145-9a15-9362d62f7c7b)

**Actual Result**:

Get access deny error when installing gti tool which had installed before.

![image](https://github.com/dotnet/sdk/assets/83067197/7f6242ec-305f-4acf-a318-43f48084da6e)

**DotNet Info**:
```
.NET SDK:

 Version:           9.0.100-alpha.1.24070.3
 Commit:            f91d4ca399
 Workload version:  9.0.100-manifests.0d37fe7a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.100-alpha.1.24070.3\

Host:
  Version:      9.0.0-alpha.1.24066.33
  Architecture: x64
  Commit:       dbb335c6ba

.NET SDKs installed:
  9.0.100-alpha.1.24070.3 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 9.0.0-alpha.1.24068.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 9.0.0-alpha.1.24066.33 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 9.0.0-alpha.1.24068.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Findings**:

The permissions for the folder path in error message not changed when running with both dotnet-sdk-8.0.101 and dotnet-sdk-9.0.100-alpha.1.24070.3. 

@dotnet-actwx-bot @dotnet/compat

**Comment by Junjun-zhao** [comment]
The same error occurred on 8.0.200-rtm.24069.25, the above verification scenarios has been updated.

**Comment by Junjun-zhao** [comment]
@marcpopMSFT @JL03-Yue This issue repro on the latest build for both .NET 9 Preview1 and .NET 8 Servicing, could you please help confirm whether it is a blocker for .NET 9 Preview1 and .NET 8 Servicing?

**Comment by JL03-Yue** [comment]
similar to #38354

**Comment by Junjun-zhao** [comment]
@marcpopMSFT @JL03-Yue According the last update, it seems this issue has been moved to .NET 9.0 servicing 9.0.1xx. Could you please help confirm what's the plan for the .NET 8.x issue https://github.com/dotnet/sdk/issues/38354 ?

**Comment by Junjun-zhao** [comment]
Verified this issue and it has been Fixed on dotnet-sdk-9.0.100-preview.7.24352.8. Hence closing this issue.
